### PR TITLE
Docker server image docs

### PR DESCRIPTION
**Pull request by Robert-Steiner**
## Issue

We want to show flower users how to start the flower server using docker.

### Description

- adds documentation how to run flower in docker

### CODE REVIEW

**Comment by Robert-Steiner** [comment]
I just saw that we rename `certificates` to `credentials` in #2663. I can update it in my pr but the SSL example won't work for Flower 1.6.0 and below. What would be the best way to handle that? Maybe adding a note?

File: doc/source/how-to-run-flower-in-docker.rst [hunk 1]
```diff
@@ -0,0 +1,143 @@
+Run Flower in Docker
```
Comment (danieljanes) [inline doc/source/how-to-run-flower-in-docker.rst:1]: @Robert-Steiner please also rename the file to reflect this:
```suggestion
Run Flower using Docker
```

File: doc/source/how-to-install-flower.rst [hunk 2]
```diff
@@ -32,6 +32,11 @@ The following command can be used to verfiy if Flower was successfully installed
 Advanced installation options
 -----------------------------
 
+Install via Docker
+~~~~~~~~~~~~~~~~~~
+
+`How to run Flower in Docker <https://flower.dev/docs/framework/how-to-run-flower-in-docker.html>`_
```
Comment (danieljanes) [inline doc/source/how-to-install-flower.rst:38]:
```suggestion
`How to run Flower using Docker <https://flower.dev/docs/framework/how-to-run-flower-in-docker.html>`_
```

File: doc/source/how-to-run-flower-in-docker.rst [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+.. code-block:: bash
+
+  $ docker run --rm -p 9091:9091 -p 9092:9092 flwr/server:1.6.0-py3.11-ubuntu22.04 \
+    --insecure
+
+The command will pull the Docker image with the tag ``1.6.0-py3.11-ubuntu22.04`` from Docker Hub.
+The tag contains the information which Flower, Python and Ubuntu is used. In this case, it
+uses Flower 1.6.0, Python 3.11 and Ubuntu 22.04. The ``--rm`` flag that tells Docker to remove
+the container after it exits.
```
Comment (danieljanes) [inline doc/source/how-to-run-flower-in-docker.rst:40]:
```suggestion
uses Flower 1.6.0, Python 3.11 and Ubuntu 22.04. The ``--rm`` flag tells Docker to remove
the container after it exits.
```

File: doc/source/how-to-run-flower-in-docker.rst [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+.. note::
+
+  By default, Flower uses a SQLite in-memory database to store its state. When using the Docker flag
```
Comment (danieljanes) [inline doc/source/how-to-run-flower-in-docker.rst:44]:
```suggestion
  By default, the Flower server keeps state in-memory. When using the Docker flag
```

File: doc/source/how-to-run-flower-in-docker.rst [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+  state in a file on your host system.
+
+The ``-p <host>:<container>`` flag tells Docker to map the ports ``9091``/``9092`` of the host to
+``9091``/``9092`` of the container, allowing you to access the Driver API on ``http://localhost:9091``
+and the Fleet API on ``http://localhost:9092``. Lastly, any flag that comes after the tag is passed
+to the Flower server. Here, we are passing the flag ``--insecure``.
+
+.. attention::
+
+  The ``--insecure`` flag enables insecure communication (not using HTTPS) and should only be used
```
Comment (danieljanes) [inline doc/source/how-to-run-flower-in-docker.rst:55]:
```suggestion
  The ``--insecure`` flag enables insecure communication (using HTTP, not HTTPS) and should only be used
```

File: doc/source/how-to-run-flower-in-docker.rst [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+Enabling SSL for secure connections
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To enable SSL, you will need a CA certificate, a server certificate and a server private key.
+
+.. note::
+  For testing purposes, you can generate your own self-signed certificates. We have a bash script
+  for your convenience. The
+  `Enable SSL connections <https://flower.dev/docs/framework/how-to-enable-ssl-connections.html#certificates>`_
+  page contains a section that will guide you through the process.
```
Comment (danieljanes) [inline doc/source/how-to-run-flower-in-docker.rst:94]: Note that the bash script we provide is for testing purposes only:
```suggestion
.. note::

  For testing purposes, you can generate your own self-signed certificates. The
  `Enable SSL connections <https://flower.dev/docs/framework/how-to-enable-ssl-connections.html#certificates>`_
  page contains a section that will guide you through the process.
```

**Review by danieljanes — CHANGES_REQUESTED** [review]

**Comment by Robert-Steiner** [comment]
> I just saw that we rename `certificates` to `credentials` in #2663. I can update it in my pr but the SSL example won't work for Flower 1.6.0 and below. What would be the best way to handle that? Maybe adding a note?

@danieljanes One solution I can think of is to keep the "--certificates" parameter but mark it as deprecated to avoid a breaking change. wdyt?

**Comment by danieljanes** [comment]
@Robert-Steiner `flower-server` is still marked "experimental", so we can still change things without deprecating them. This will change soon, but for now it's ok to just change it.

**Comment by Robert-Steiner** [comment]
> @Robert-Steiner `flower-server` is still marked "experimental", so we can still change things without deprecating them. This will change soon, but for now it's ok to just change it.

Alright, that makes sense, then I'll leave it as it is.

**Review by tanertopal — APPROVED** [review]
lgtm!

**Pull request merged**